src/planning/src/planning/cone_coloring.cpp

```diff
@@ -0,0 +1,137 @@
+#include "planning/cone_coloring.hpp"
+#include <iostream>
+
+ConeColoring::ConeColoring(double gain_angle, double gain_distance, double gain_ncones,
+ double exponent_1, double exponent_2, double cost_max ) : angle_gain(gain_angle),
+ distance_gain(gain_distance), ncones_gain(gain_ncones), exponent1(exponent_1),
+ exponent2(exponent_2), max_cost(cost_max) {}
+
+double ConeColoring::angle_and_norm(Cone *possible_next_cone, bool side, double &norm) {
+    std::vector<Cone *> current_cones = side ? current_left_cones : current_right_cones;
```

Comment:
Use a pointer to the data structure instead of this, as you are always copying it like this (look into shared_ptr)